src/buffer.rs

```diff
@@ -42,17 +32,12 @@ impl<T> Buffer<T> for &mut [T] {
     }
 }
 
-/// Implements [`Buffer`] around the a slice of bytes.
-///
-/// `Result` is a `usize` indicating how many bytes were written.
-impl<T, const N: usize> Buffer<T> for &mut [T; N] {
+impl<T, const N: usize> private::Sealed<T> for &mut [T; N] {
```

Comment:
Pretty sure we can still implement the buffer trait here so we don't have to specify the impls twice.

Reply:
I figured we should make the contents of the `Buffer` trait private, at least for now, so we can evolve it as needed.

Reply:
Oh right sorry I forgot that sealed traits prevent you from implementing them in your own types but not from using them. Clever idea!